modules/search/pipeline.go

```diff
@@ -0,0 +1,88 @@
+package search
+
+import (
+	"strings"
+	"time"
+
+	"github.com/grafana/tempo/pkg/tempofb"
+	"github.com/grafana/tempo/pkg/tempopb"
+)
+
+type tracefilter func(header *tempofb.SearchData) (matches bool)
+
+type Pipeline struct {
```

Comment:
nit: name

Predicate? Matcher? ComparerThing?

Reply:
Naming was inspired by Loki:  https://github.com/grafana/loki/blob/main/pkg/logql/log/pipeline.go  Might be worth considering as more complex search capabilities are added.